packages/astro/src/runtime/server/index.ts

```diff
@@ -730,28 +731,48 @@ export async function renderPage(
 	const factoryReturnValue = await componentFactory(result, props, children);
 
 	if (isAstroComponent(factoryReturnValue)) {
-		iterable = renderAstroComponent(factoryReturnValue);
-		let stream = new ReadableStream({
-			start(controller) {
-				async function read() {
-					let i = 0;
-					for await (const chunk of iterable) {
-						let html = chunk.toString();
-						if (i === 0) {
-							if (!/<!doctype html/i.test(html)) {
-								controller.enqueue(encoder.encode('<!DOCTYPE html>\n'));
+		let iterable = renderAstroComponent(factoryReturnValue);
+		let init = result.response;
+		let headers = new Headers(init.headers);
+		let body: BodyInit;
+		if (streaming) {
```

Comment:
Bah, I wish CF supported `ReadableStream`. I'm still on a mission to reduce forking code paths, so in that spirit I'd love to use `ReadableStream` in both cases and then just pipe it to a string as a smaller, additional step in the non-streaming code path.

I can't think of any good way to remove the copy-paste code between the two `if-else` code paths, so I guess this might not be actionable. Just venting I guess :)

Reply:
That was actually my first solution!  We can't use `ReadableStream` in CF at all unfortunately, the constructor throws an error immediately.  Even worse, the error is basically silent if you don't add an extra try/catch around the entire SSR handler and logs are accessible yet for Cloudflare Pages Functions 🙃